libsolidity/ast/Types.cpp

```diff
@@ -2698,6 +2708,96 @@ Type const* TupleType::mobileType() const
 	return TypeProvider::tuple(move(mobiles));
 }
 
+BoolResult InlineArrayType::isImplicitlyConvertibleTo(Type const& _other) const
+{
+	auto arrayType = dynamic_cast<ArrayType const*>(&_other);
+
+	if (!arrayType || arrayType->isByteArrayOrString())
+		return BoolResult::err("Array literal can not be converted to byte array or string.");
+	else
+	{
+		if (!arrayType->isDynamicallySized() && arrayType->length() != components().size())
+			return BoolResult::err(
+				"Number of components in array literal (" + to_string(components().size()) + ") " +
+				"does not match array size (" + to_string(arrayType->length().convert_to<unsigned>()) + ").");
+
+		for (Type const* c: components())
+		{
+			BoolResult result = c->isImplicitlyConvertibleTo(*arrayType->baseType());
+			if (!result)
+				if (!result)
+					return BoolResult::err(
+						"Invalid conversion from " + c->toString(false) +
+						" to " + arrayType->baseType()->toString(false) + "."
+						+ (result.message().empty() ? "" : " ") + result.message() );
```

Comment:
```suggestion
						+ (result.message().empty() ? "" : " ") + result.message()
					);
```

(and actually we break after each `+`, but I think in this case we can leave it like that

Reply:
OK, I will just move '+' from the beginning of the last line to the end of the second to last line.